### PR TITLE
Implement board properties

### PR DESCRIPTION
**Pull request by fanneykristjans**
# Pull request
## :heavy_exclamation_mark: Check list
- [X] Did you follow the TDD protocol?  
- [X] Did your code build?  
- [X] Did your code pass through Travis?  
- [X] Did you write a description about your branch and its purpose?
## :heavy_exclamation_mark: Description and purpose of the branch:

List of Board functionalities/properties that have been implemented:
- initialization and constructor
- get and set functions, with exceptions thrown if wrong input and ignored if some symbol other than X/O is set
- has three in row, i.e. if someone has won
- is full, i.e. if the game is over (doesn't imply if it was a tie or if someone won)

@reykjalin @thoreyjona @Herdismaria @sarakristjans who would like to review this endless branch for merging? 💯

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/PowerRangersHug/TicTacToe/pull/10" target="_blank">ReviewNinja

File: src/main/java/is/TicTacToe/data/Board.java [hunk 1]
```diff
@@ -2,15 +2,107 @@
...
-    public String getSymbol(int x, int y) { return ""; }
-    public void setSymbol(int x, int y, String symbol) {  }
+    /**
+	* Get the current symbol at the (x,y) grid position.
+ 	* @param int x, int y
+ 	* @return a symbol
+ 	*/
+    public String getSymbol(int x, int y) 
+    {
+    	if ((x < 0 || x > 2) && (y < 0 || y > 2))
```
Comment (Herdismaria) [inline src/main/java/is/TicTacToe/data/Board.java]: Should be || instead of &&

Comment (fanneykristjans) [reply src/main/java/is/TicTacToe/data/Board.java]: Oh thats right! I will fix this.

**Review by Herdismaria — CHANGES_REQUESTED** [review]
Fix line 24

**Review by Herdismaria — DISMISSED** [review]

File: src/main/java/is/TicTacToe/data/Board.java [hunk 2]
```diff
@@ -2,15 +2,107 @@
...
-    public boolean hasThreeInRow() { return false; }
-    public boolean isFull() { return false; }
+    /**
+	* Set the current symbol at the (x,y) grid position.
+ 	* @param int x, int y
+ 	* @return
+ 	*/
+    public void setSymbol(int x, int y, String symbol)
+    {
+    	if ((x < 0 || x > 2) && (y < 0 || y > 2))
```
Comment (Herdismaria) [inline src/main/java/is/TicTacToe/data/Board.java]: Change && into ||

File: src/main/java/is/TicTacToe/data/Board.java [hunk 3]
```diff
@@ -2,15 +2,107 @@
...
+
+    private boolean diagonalsThreeInRow()
+    {
+    	return (grid[0][0] != "") && (grid[0][0] == grid[1][1]) && (grid[1][1] == grid[2][2]);
+    }
+
+    private boolean hasThreeInCurrentRow(int i)
+    {
+    	return (grid[i][0] != "") && (grid[i][0] == grid[i][1]) && (grid[i][1] == grid[i][2]);
+    }
```
Comment (Herdismaria) [inline src/main/java/is/TicTacToe/data/Board.java:113]: Need to check the columns too

File: src/test/java/is/TicTacToe/data/BoardTest.java [hunk 4]
```diff
@@ -1,7 +1,71 @@
...
+	}
+
+	@Test
+    public void testBoardGetWrongField()
+    {
+    	Board testBoard = new Board();
+    	thrown.expect(IndexOutOfBoundsException.class);
+		thrown.expectMessage("Index out of bounds!");
+    	testBoard.getSymbol(3,3);
+	}
```
Comment (Herdismaria) [inline src/test/java/is/TicTacToe/data/BoardTest.java]: What is this INDENTATION?

**Review by Herdismaria — CHANGES_REQUESTED** [review]

**Review by reykjalin — DISMISSED** [review]
Fix indentation.
Use spaces :angry:

File: src/test/java/is/TicTacToe/data/BoardTest.java (hunk 4 above)
Comment (reykjalin) [reply src/test/java/is/TicTacToe/data/BoardTest.java]: Use spaces :angry:

**Review by Herdismaria — DISMISSED** [review]

File: src/main/java/is/TicTacToe/data/Board.java [hunk 5]
```diff
@@ -2,15 +2,113 @@
...
+    /**
+    * Checks if the board is in a winning state, i.e.
+    * a symbol occurs three times in a row on the board.
+    * @param
+    * @return true/false whether the board has three in row or not.
+    */
+    public boolean hasThreeInRow()
+    {
+        if (diagonalsThreeInRow())
+        	return true;
```
Comment (reykjalin) [inline src/main/java/is/TicTacToe/data/Board.java]: Fix indentation here

File: src/test/java/is/TicTacToe/data/BoardTest.java [hunk 6]
```diff
@@ -1,7 +1,71 @@
...
 public class BoardTest {
+    @Rule public final ExpectedException thrown = ExpectedException.none();
+
+
+    @Test
+    public void testBoardSetAndGet()
+    {
+        Board testBoard = new Board();
+        testBoard.setSymbol(1,1,"X");
+       assertEquals("X", testBoard.getSymbol(1,1));
```
Comment (reykjalin) [inline src/test/java/is/TicTacToe/data/BoardTest.java]: Fix indentation

File: src/test/java/is/TicTacToe/data/BoardTest.java [hunk 7]
```diff
@@ -1,7 +1,71 @@
...
+    }
+
+    @Test
+    public void testBoardSetAndGetWrongSymbol()
+    {
+        Board testBoard = new Board();
+        testBoard.setSymbol(1,1,"BLA");
+       assertEquals("", testBoard.getSymbol(1,1));
```
Comment (reykjalin) [inline src/test/java/is/TicTacToe/data/BoardTest.java]: Fix indentation

**Review by reykjalin — CHANGES_REQUESTED** [review]
Fix indentation errors

**Review by reykjalin — APPROVED** [review]
Approved. Finally! :stuck_out_tongue:

**Pull request merged**